<launch>  

  <group ns="lgn2v1">
    <node respawn="true" name="converter" pkg="lgn2v1" type="lgn2v1-node">
      <remap from="in" to="/smallbot/front" />
      <remap from="dfocus" to="/smallbot/dpantilt" />
    </node>
  </group>
  
  <group ns="color_filter">
    <node respawn="true" name="filter" pkg="color_filter" type="color-filter-hsv-node">
      <remap from="in" to="/lgn2v1/lgn" />
    </node>
  </group>

  <group ns="focus">
    <node respawn="true" name="locus" pkg="focus" type="locus.py">
      <remap from="in/compressed" to="/color_filter/filtered/compressed"/>
      <remap from="shift" to="/lgn2v1/convert"/>
    </node>
  </group>
  
  
  <group ns="animat">
    <group ns="vision">
      <node respawn="true" name="vision" pkg="animat" type="vision.py">
	<remap from="raw/compressed" to="/smallbot/front/compressed"/>
	<remap from="in/compressed" to="/color_filter/filtered/compressed"/>
	<remap from="focus" to="/smallbot/camerapos"/>
	<remap from="set_focus" to="/smallbot/pantilt"/>
	<remap from="hsv" to="/color_filter/params"/>
      </node>
    </group>
    
    <group ns="physiology">
      <node respawn="true" name="physiology" pkg="animat" type="physiology.py">
      </node>
    </group>
    
    <group ns="embodiment">
      <node respawn="true" name="embodiment" pkg="animat" type="embodiment.py">
	<remap from="cmd_vel" to="/smallbot/cmd_vel"/>
	<remap from="area" to="/world/area"/>
	<remap from="physio" to="/animat/physiology/command"/>
      </node>
    </group>

    <node name="physio" pkg="rqt_plot" type="rqt_plot" args="/animat/physiology/hydration /animat/physiology/glycemia">
    </node>
    
    <node name="animat_teleop" pkg="animat" type="animat_teleop.py" output="screen" launch-prefix="xterm -e">
      <remap from="body" to="/animat/embodiment/command"/>
      <remap from="vision" to="/animat/vision/command"/>
    </node>
  </group>

  <node respawn="true" name="hardwired" pkg="animat" type="hardwired.py">
    <remap from="hydration"   to="/animat/physiology/hydration"/>
    <remap from="glycemia"    to="/animat/physiology/glycemia"/>
    <remap from="fovea"       to="/animat/vision/fovea"/>
    <remap from="attention"   to="/animat/vision/command"/>
    <remap from="ingest"      to="/animat/embodiment/command"/>
    <remap from="focus"       to="/smallbot/camerapos"/>
    <remap from="cmd_vel"     to="/smallbot/cmd_vel"/>
    <remap from="pantilt"     to="/smallbot/pantilt"/>
  </node>
  
  <node name="gaze_view" pkg="image_view" type="image_view">
    <param name="image_transport" value="compressed"/>
    <remap from="image" to="/animat/vision/gaze"/>
  </node>

  <node name="vision_param"     pkg="dynamic_reconfigure" type="dynparam" args="load /animat/vision/vision $(find animat)/params/hardwired-controler-vision.yaml"/>
  <node name="embodiment_param" pkg="dynamic_reconfigure" type="dynparam" args="load /animat/embodiment/embodiment $(find animat)/params/hardwired-controler-embodiment.yaml"/>
  <node name="locus_param"      pkg="dynamic_reconfigure" type="dynparam" args="load /focus/locus $(find animat)/params/hardwired-controler-locus.yaml"/>
  <node name="lgn2v1_param"     pkg="dynamic_reconfigure" type="dynparam" args="load /lgn2v1/converter $(find animat)/params/hardwired-controler-lgn2v1.yaml"/>
  <node name="physiology_param"     pkg="dynamic_reconfigure" type="dynparam" args="load /animat/physiology/physiology $(find animat)/params/hardwired-controler-physiology.yaml"/>

</launch>
